app-emulation/wine-proton/wine-proton-7.0.9999.ebuild

```diff
@@ -0,0 +1,319 @@
+# Copyright 2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+MULTILIB_COMPAT=( abi_x86_{32,64} )
+PYTHON_COMPAT=( python3_{8..11} )
+inherit autotools flag-o-matic multilib multilib-build \
+	python-any-r1 readme.gentoo-r1 toolchain-funcs wrapper
+
+WINE_GECKO=2.47.3
+WINE_MONO=7.3.1
+WINE_PV=$(ver_rs 2 -)
+
+if [[ ${PV} == *9999 ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/ValveSoftware/wine.git"
+	EGIT_BRANCH="experimental_$(ver_cut 1-2)"
+else
+	SRC_URI="https://github.com/ValveSoftware/wine/archive/refs/tags/proton-wine-${WINE_PV}.tar.gz"
+	S="${WORKDIR}/${PN}-wine-${WINE_PV}"
+	KEYWORDS="-* ~amd64 ~x86"
+fi
+
+DESCRIPTION="Valve Software's fork of Wine"
+HOMEPAGE="https://github.com/ValveSoftware/wine/"
+
+LICENSE="LGPL-2.1+ BSD-2 IJG MIT ZLIB gsm libpng2 libtiff"
+SLOT="${PV}"
+IUSE="
+	+abi_x86_32 +abi_x86_64 +alsa crossdev-mingw custom-cflags debug
+	+fontconfig +gecko +gstreamer llvm-libunwind +mono nls openal
+	osmesa perl pulseaudio +sdl selinux +ssl udev udisks +unwind usb
+	v4l +vkd3d +xcomposite xinerama"
+
+# tests are non-trivial to run, can hang easily, don't play well with
+# sandbox, and several need real opengl/vulkan or network access
```

Comment:
I have no doubt you've looked into this, but for my own curiosity: is there a subset we can run? (doesn't look like there's some nice target for it, but possibly could construct a list of some of the more basic DLLs and see what happens)

Avoiding graphics tests should be easier but I guess there'd be a struggle with avoiding network tests. Possible that even some of the simpler "basic" DLL tests might end up trying to do something with the network.

Reply:
Yeah it would probably be easier to run tests only for specific dlls than try to run the top level `make test` and try to skip specific tests (need to figure out which test.ok to `touch` and there's hundreds of files), albeit list of dlls is massive too.

Running the tests normally already failed a lot, and it hanged on a few different occasions, sometime giving me a window prompt to interact with that just staying there (that wasn't the only kind of window, but at bare minimum would also need to disable the crash handler with regedit given that means hanging if it segfaults).

With sandbox I had a lot of strange memory write errors that I haven't figured out yet though, not that I messed with it much within the sandbox.

Reply:
I'll keep trying to get a working subset on my TODO anyhow, won't do this now but may revisit later.

Reply:
No worries, thanks for explaining!